tests/Makefile

```diff
@@ -71,6 +71,9 @@ all: $(PROGS) $(UTILS) $(PLAT_UTILS) $(FUNCS) $(SPECIAL_FUNCS) $(TEST_TARGET)
 
 tests_only: $(PROGS) $(TEST_TARGET)
 
+GITIGN = .gitignore
+MKGITIGN = [ -f "$(GITIGN)" ] || echo "$(GITIGN)" > $(GITIGN); grep -q '^\$@$$' $(GITIGN) || echo "$@" >> $(GITIGN)
```

Comment:
What does this do, and why?  It looks like it's echoing something into the end of (`>>`) the `.gitignore` file, but that doesn't make sense, because `.gitignore` is supposed to be version-controlled. `make test` shouldn't *modify* a version-controlled file, should it?